src/main.js

```diff
@@ -218,49 +218,58 @@ app.whenReady().then(async () => {
 		}
 	})
 
+	const trustedFingerprints = getAppConfig('trustedFingerprints') ?? []
 	app.on('certificate-error', (event, webContents, url, error, certificate, callback) => {
 		event.preventDefault()
 
 		if (isLinux) {
-			return callback(process.env.NODE_ENV !== 'production')
+			if (trustedFingerprints.includes(certificate.fingerprint)) {
+				callback(true)
+			} else {
+				dialog.showMessageBox(mainWindow, {
+					type: 'warning',
+					title: 'Security Warning',
+					detail:
+						[
+							`Error: ${error}`,
+							'',
+							`Subject: ${certificate.subjectName}`,
+							'',
+							`Issuer: ${certificate.issuerName ?? 'UNKNOWN'}`,
+							`- Organisations: ${certificate.issuer.organizations.join(', ')}`,
+							`- Organisation units: ${certificate.issuer.organizationUnits.join(', ')}`,
+							`- Country: ${certificate.issuer.country}`,
+							`- State: ${certificate.issuer.state}`,
+							`- Locality: ${certificate.issuer.locality}`,
+							'',
+							`Fingerprint: ${certificate.fingerprint}`,
+							'',
+							`Valid from: ${new Date(certificate.validStart * 1_000).toLocaleDateString()}`,
+							`Valid until: ${new Date(certificate.validExpiry * 1_000).toLocaleDateString()}`,
+							'',
+							'Do you trust this certificate?',
+						].join('\n'),
+					buttons: ['Yes', 'Cancel'],
+				}).then((res) => {
+					if (res.response === 0) {
+						trustedFingerprints.push(certificate.fingerprint)
+						setAppConfig('trustedFingerprints', trustedFingerprints)
```

Comment:
At this moment there is a chance that the certificate was already added from another dialog.
Let's only add a new certificate if it is not added yet in parallel.

Reply:
Maybe this is `setAppConfig` problem, but it only saves the config when the set value is actually new. And it compares it by reference. Setting the same array with a new pushed certificate has no effect.

As a result, it only works when `trustedFingerprints` was not defined before.

Having an empty list, or a list with old fingerprints prevents from adding a new one.

```json
{
  "trustedFingerprints": [],
  "//": "or",
  "trustedFingerprints": ["...old fingerprint..."]
}
```

Reply:
A simple option here is creating a new array instead of a push.

Alternative - adjust `setAppConfig` to compare values by value, not reference. For example, via `JSON.stringify`.